src/App.jsx

```diff
@@ -769,7 +769,7 @@ const App = () => {
         setUserInputs(updatedUserInputs);
     };
 
-    const handleRowClick = (rowKey) => {
+    const handleRowClick = (rowKey) => { 
```

Comment:
I haven't reviewed the internals of this function but it is obvious that it has to do with the table's rows. If you get to separate this component from the App then this should go away as well, leaving only the relevant things in the App (things that should be potentially global in the App).

In the case you want to update the "global" state (in this case residing in the App), you could use callbacks that you can pass as properties. The App shouldn't "care" of what is happening in the Table, but the Table could communicate information back to the tree. (This could be handled even easier in nested components with the use of context)